janus_core/calculations/descriptors.py

```diff
@@ -104,18 +110,23 @@ def __init__(
             if not self.struct.calc:
                 raise ValueError("Please attach a calculator to `struct`.")
 
-        # Configure logging
-        log_kwargs.setdefault("name", __name__)
-        self.logger = config_logger(**log_kwargs)
-        self.tracker = config_tracker(self.logger, **tracker_kwargs)
-
         # Set output file
         FileNameMixin.__init__(self, struct, None)
         self.write_kwargs.setdefault(
             "filename",
             self._build_filename("descriptors.extxyz").absolute(),
         )
 
+        # Configure logging
+        if attach_logger:
+            self.log_kwargs.setdefault(
+                "filename",
+                self._build_filename("log.yml").absolute(),
+            )
+        self.log_kwargs.setdefault("name", __name__)
+        self.logger = config_logger(**self.log_kwargs)
+        self.tracker = config_tracker(self.logger, **tracker_kwargs)
+
```

Comment:
Is there some way to avoid repeating this pattern in the various classes that use it? I suppose it'll end up similar to the ```FileNameMixin``` and there will have to be some repetition.

Reply:
`FileNameMixin` could probably absorb this as a method taking `attach_logger` as an argument. Or some other mixin `LoggerMixin`